src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -936,11 +957,50 @@ Coordination::ReconfigResponse ZooKeeper::reconfig(
 }
 
 
+ZooKeeperPtr ZooKeeper::createWithoutKillingPreviousSessions(const ZooKeeperArgs & args_)
+{
+    return std::shared_ptr<ZooKeeper>(new ZooKeeper(args_));
+}
```

Comment:
```suggestion
template <typename... Args>
ZooKeeperPtr ZooKeeper::createWithoutKillingPreviousSessions(Args &&.... args)
{
    return std::shared_ptr<ZooKeeper>(std::forward<Args>(args)...);
}
```
maybe something like this so we don't have to define this function for each constructor?